libs/clients/health-directorate/src/lib/healthDirectorateClient.service.ts

```diff
@@ -137,6 +150,57 @@ export class HealthDirectorateClientService {
     return result
   }
 
+  async getHealthCareLicensesForWorkPermit(
+    auth: Auth,
+  ): Promise<StarfsleyfiUmsoknStarfsleyfi[]> {
+    const licenses = await this.umsoknStarfsleyfiApiWith(
+      auth,
+    ).umsoknStarfsleyfiStarfsleyfiGet()
+
+    return licenses
+  }
+
+  async getHealthCareWorkPermitEducationInfo(
+    auth: Auth,
+  ): Promise<NamsUpplysingar[]> {
+    const educationInfo = await this.umsoknStarfsleyfiApiWith(
+      auth,
+    ).umsoknStarfsleyfiNamsUpplysGet()
+
+    return educationInfo
+  }
+
+  async submitApplicationHealthcareWorkPermit(
+    auth: User,
+    request: HealthcareWorkPermitRequest,
+  ): Promise<UtbuaStarfsleyfiSkjalResponse[]> {
+    const items = await this.umsoknStarfsleyfiApiWith(
+      auth,
+    ).umsoknStarfsleyfiUtbuaSkjalPost({
+      utbuaStarfsleyfiSkjalRequest: {
+        name: request.name,
+        dateOfBirth: format(new Date(request.dateOfBirth), 'dd.MM.yyyy'),
+        citizenship: request.citizenship,
+        email: request.email,
+        phoneNo: request.phone,
+        idProfession: request.idProfession,
+        education: request.education,
+      },
+    })
+
+    if (!items || !Array.isArray(items) || items.length === 0) {
+      throw new Error('No items returned')
+    }
+
+    for (const item of items) {
+      if (!item.base64String) {
+        throw new Error('Empty file in one of the items')
+      }
+    }
```

Comment:
Might be more concise to use `Array.some()`, but ymmv

```suggestion
    if (items.some((item) => !item.base64String)) {
      throw new Error('Empty file in one of the items')
    }
```

Reply:
I'm indifferent to either style. I changed it to the functional way.